src/NServiceBus.AmazonSQS/Configure/TransportInfrastructure.cs

```diff
@@ -71,6 +71,13 @@ public override TransportSendInfrastructure ConfigureSendInfrastructure()
                 () => Task.FromResult(StartupCheckResult.Success));
         }
 
+        public override Task Stop()
+        {
+            sqsClient.Dispose();
+            s3Client.Dispose();
+            return base.Stop();
+        }
```

Comment:
any chance `Stop` is called by Core, I suppose, and constructor initialization is not fully completed?
Or put in another way: what happens if the `TransportInfracture.ctor()` throws? Can `sqsClient` and/or `s3Client` be null and `Stop` still be called?

Reply:
When the ctor throws this happens during Endpoint.Start() which means you have no endpoint instance you can stop